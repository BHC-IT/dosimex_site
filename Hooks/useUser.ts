import { useState, useEffect } from 'react';

export interface IUser {
	username: string,
	password: string,
}


const useUser = () => {

	const [token, setToken] = useState<string | null>(null)

	useEffect(() => {
		setToken(localStorage.getItem('access_token'))
	}, []);

	return token;
}

export default useUser;
