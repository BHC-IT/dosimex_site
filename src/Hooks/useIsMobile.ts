import * as React from 'react'

import { isMobile, isTablet } from 'react-device-detect'

export const useIsMobile = <T>(styles: (m: boolean) => T): T | null => {
	// @ts-ignore - React hooks compatibility with React 18
	const [style, setStyle] = React.useState<T | null>(styles(isMobile))

	// @ts-ignore - React hooks compatibility with React 18
	React.useEffect(() => {
		setStyle(null)
		setTimeout(() => setStyle(styles(isMobile)), 10)
	}, [])

	return style
}

export const useMobile = (): boolean | null => {
	// @ts-ignore - React hooks compatibility with React 18
	const [mobile, setMobile] = React.useState<boolean | null>(null)

	// @ts-ignore - React hooks compatibility with React 18
	React.useEffect(() => {
		setTimeout(() => setMobile(isMobile), 10)
	}, [])

	return mobile
}

export const useTablet = (): boolean | null => {
	// @ts-ignore - React hooks compatibility with React 18
	const [tablet, setTablet] = React.useState<boolean | null>(null)

	// @ts-ignore - React hooks compatibility with React 18
	React.useEffect(() => {
		setTimeout(() => setTablet(isTablet), 10)
	}, [])

	return tablet
}

export const useDeviceType = (): { isMobile: boolean | null; isTablet: boolean | null } => {
	const mobile = useMobile()
	const tablet = useTablet()

	return { isMobile: mobile, isTablet: tablet }
}
