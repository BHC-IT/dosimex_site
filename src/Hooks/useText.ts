import { useRouter } from 'next/router'

import debug from '../lang/debug'
import en from '../lang/en'
import fr from '../lang/fr'
import ILang from '../lang/interface'

export const text : {[$:string]: ILang} = { fr, en, debug }

export const useText = (page: keyof ILang): ILang[keyof ILang] | null => {
	const route = useRouter()

	if (route.locale === 'debug') {
		return text[route.locale]?.[page] ?? null
	} else if (route?.locale) {
		const locale = route.locale.slice(0, 2)
		return text[locale]?.[page] ?? null
	}
	return null
}
