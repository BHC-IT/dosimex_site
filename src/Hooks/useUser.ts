import { useState, useEffect } from 'react';


export interface IUser {
	username: string,
	password: string,
}

const remove = (setToken: Function) => {
	localStorage.removeItem('access_token')
	localStorage.removeItem('exp_token')
	setToken(null)
}

const useUser = () => {

	const [token, setToken] = useState<string | null>(null)

	useEffect(() => {
		setToken(localStorage.getItem('access_token'))
		const exp = Number(localStorage.getItem('exp_token'))

		if (!exp) {
			remove(setToken)
			return;
		}
		if (exp < Date.now())
			remove(setToken)
		else
			setTimeout(() => remove(setToken), exp - Date.now())
	}, []);

	return token;
}

export default useUser;
