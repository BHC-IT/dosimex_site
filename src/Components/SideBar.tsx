import Image from 'next/image'
import Link from 'next/link'
import { useRouter } from 'next/router'
import React from 'react'
import { slide as Menu } from 'react-burger-menu'

import { useText } from '../Hooks/useText'

import Button from './Button'
import ItemNavbar from './ItemNavbar'
import LanguageSwitch from './LanguageSwitch'

interface IPage {
	route: string
}

interface IProps {
	text?: {
		items: string[]
		button: string
	}
}

const pages: IPage[] = [
	{ route: 'Software' },
	{ route: 'Training' },
	{ route: 'Manuals' },
	{ route: 'About' },
	{ route: 'Contact' },
]

interface IRenderNavProps {
	text?: {
		items: string[]
		button: string
	} | null
}

const RenderNav = ({ text }: IRenderNavProps) => {
	// Only render navigation items if text.items exists and has content
	if (!text?.items || text.items.length === 0) {
		return null
	}

	return (
		<>
			{pages.map((page: IPage, i: number) => {
				// Only render if the text item exists
				if (!text.items[i]) {
					return null
				}

				return (
					<li
						className='menu-item'
						key={text.items[i]}
					>
						<ItemNavbar
							name={text.items[i]}
							route={page.route}
						/>
					</li>
				)
			})}
		</>
	)
}

const ratio = 0.5

const SideBar = (props: IProps) => {
	const text = useText('Navbar')
	const router = useRouter()

	return React.createElement(
		Menu as React.ComponentType<Record<string, unknown>>,
		{
			right: true,
			...props,
		},
		React.createElement(
			'ul',
			null,
			React.createElement(
				'li',
				{ style: { cursor: 'pointer' } },
				React.createElement(
					Link,
					{ href: '/' },
					React.createElement(Image, {
						src: '/Images/logo_dosimex_new.webp',
						alt: 'logo dosimex',
						width: 212 * ratio,
						height: 44 * ratio,
						loading: 'lazy',
						quality: 40,
					}),
				),
			),
			React.createElement(RenderNav, { text }),
			React.createElement(LanguageSwitch, {
				route: router.pathname,
				language: router.locale,
			}),
		),
		React.createElement(Button, {
			name: text.button,
			route: 'Product',
		}),
	)
}

export default SideBar
