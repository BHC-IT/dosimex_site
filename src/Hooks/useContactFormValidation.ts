import emailjs from '@emailjs/browser'
import { useState, useCallback } from 'react'
import { isPossiblePhoneNumber } from 'react-phone-number-input'
import { toast } from 'react-toastify'

import { IValidator } from '../Components/Input'

// Email validation regex
const EMAIL_REGEX =
	/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/

// Validation functions
export const isInputValid = (value: string): boolean => value.trim() !== ''
export const isEmailValid = (value: string): boolean => EMAIL_REGEX.test(value)
export const isPhoneValidNumber = (phone: string | undefined): boolean =>
	!phone || isPossiblePhoneNumber(phone)

// Validator factory functions
export const createRequiredValidator = (errorMessage: string): IValidator => ({
	validationFunction: value => isInputValid(value),
	errorMessage,
})

export const createEmailValidator = (errorMessage: string): IValidator => ({
	validationFunction: value => isEmailValid(value),
	errorMessage,
})

export interface IContactFormValidation {
	// Form field states
	name: string | null
	email: string | null
	phone: string | undefined
	enterprise: string | null
	address: string | null
	subject: string | null
	message: string | null

	// Validation states
	nameValid: boolean
	phoneError: string | null
	isLoading: boolean

	// Form handlers
	setName: (value: string) => void
	setEmail: (value: string) => void
	setPhone: (value: string | undefined) => void
	setEnterprise: (value: string) => void
	setAddress: (value: string) => void
	setSubject: (value: string) => void
	setMessage: (value: string) => void
	setNameValid: (valid: boolean) => void
	setIsLoading: (loading: boolean) => void

	// Validation methods
	validatePhone: () => void
	validateForm: () => boolean
	resetForm: () => void
	handleSubmit: (e: React.FormEvent) => void

	// Validator configurations
	getNameValidators: (errorMessage: string) => IValidator[]
	getEmailValidators: (errorEmail: string[]) => IValidator[]
	getAddressValidators: (errorMessage: string) => IValidator[]
}

// Validation helper functions
const validateName = (name: string | null): boolean => {
	return name !== null && name.trim() !== ''
}

const validateEmail = (email: string | null): boolean => {
	if (!email || email.trim() === '') return false
	const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
	return emailRegex.test(email)
}

const validateMessage = (message: string | null): boolean => {
	return message !== null && message.trim() !== ''
}

// Custom hook for form state management
const useFormState = () => {
	const [name, setName] = useState<string | null>(null)
	const [email, setEmail] = useState<string | null>(null)
	const [phone, setPhone] = useState<string | undefined>(undefined)
	const [enterprise, setEnterprise] = useState<string | null>(null)
	const [address, setAddress] = useState<string | null>(null)
	const [subject, setSubject] = useState<string | null>(null)
	const [message, setMessage] = useState<string | null>(null)
	const [nameValid, setNameValid] = useState(false)
	const [isLoading, setIsLoading] = useState(false)

	return {
		name, setName,
		email, setEmail,
		phone, setPhone,
		enterprise, setEnterprise,
		address, setAddress,
		subject, setSubject,
		message, setMessage,
		nameValid, setNameValid,
		isLoading, setIsLoading,
	}
}

// Form validation logic
const useFormValidation = (nameValid: boolean, phone: string | undefined) => {
	const checkInputs = useCallback((): boolean => {
		return nameValid && (phone ? isPossiblePhoneNumber(phone) : true)
	}, [nameValid, phone])

	const validateForm = useCallback(() => checkInputs(), [checkInputs])

	const validatePhone = useCallback(() => {
		// Phone validation is done in checkInputs
	}, [])

	return { checkInputs, validateForm, validatePhone }
}

// Form reset functionality
const useFormReset = (
	setName: (value: string | null) => void,
	setEmail: (value: string | null) => void,
	setPhone: (value: string | undefined) => void,
	setEnterprise: (value: string | null) => void,
	setAddress: (value: string | null) => void,
	setSubject: (value: string | null) => void,
	setMessage: (value: string | null) => void,
	setNameValid: (valid: boolean) => void,
	setIsLoading: (loading: boolean) => void,
) => {
	const resetForm = useCallback(() => {
		setName(null)
		setEmail(null)
		setPhone(undefined)
		setEnterprise(null)
		setAddress(null)
		setSubject(null)
		setMessage(null)
		setNameValid(false)
		setIsLoading(false)
	}, [setName, setEmail, setPhone, setEnterprise, setAddress, setSubject, setMessage, setNameValid, setIsLoading])

	return { resetForm }
}

// Email sending functionality
const useEmailSending = (
	name: string | null,
	email: string | null,
	phone: string | undefined,
	enterprise: string | null,
	address: string | null,
	subject: string | null,
	message: string | null,
	text: Record<string, string | string[]> | undefined,
	setIsLoading: (loading: boolean) => void,
	resetForm: () => void,
) => {
	const sendEmail = useCallback((): void => {
		sendContactEmail({
			name, email, phone, enterprise, address, subject, message,
			text, setIsLoading, resetForm,
		})
	}, [name, email, phone, enterprise, address, subject, message, text, setIsLoading, resetForm])

	return { sendEmail }
}

// Validator creation functions
const useValidatorCreation = (
	name: string | null,
	email: string | null,
	message: string | null,
) => {
	const getNameValidators = useCallback((errorName: string): IValidator[] => [
		{
			validationFunction: () => validateName(name),
			errorMessage: errorName || 'Name is required',
		},
	], [name])

	const getEmailValidators = useCallback((errorEmail: string[]): IValidator[] => [
		{
			validationFunction: () => email !== null && email.trim() !== '',
			errorMessage: errorEmail[0] || 'Email is required',
		},
		{
			validationFunction: () => validateEmail(email),
			errorMessage: errorEmail[1] || 'Email format is invalid',
		},
	], [email])

	const getAddressValidators = useCallback((errorMessage: string): IValidator[] => [
		{
			validationFunction: () => validateMessage(message),
			errorMessage: errorMessage || 'Message is required',
		},
	], [message])

	return { getNameValidators, getEmailValidators, getAddressValidators }
}

// Form submission handling
const useFormSubmission = (checkInputs: () => boolean, sendEmail: () => void) => {
	const handleSubmit = useCallback((e: React.FormEvent) => {
		e.preventDefault()
		if (checkInputs()) {
			sendEmail()
		}
	}, [checkInputs, sendEmail])

	return { handleSubmit }
}

export const useContactFormValidation = (text?: Record<string, string | string[]>): IContactFormValidation => {
	const formState = useFormState()
	const formValidation = useFormValidation(formState.nameValid, formState.phone)
	const { resetForm } = useFormReset(
		formState.setName,
		formState.setEmail,
		formState.setPhone,
		formState.setEnterprise,
		formState.setAddress,
		formState.setSubject,
		formState.setMessage,
		formState.setNameValid,
		formState.setIsLoading,
	)
	const { sendEmail } = useEmailSending(
		formState.name,
		formState.email,
		formState.phone,
		formState.enterprise,
		formState.address,
		formState.subject,
		formState.message,
		text,
		formState.setIsLoading,
		resetForm,
	)
	const validatorCreation = useValidatorCreation(formState.name, formState.email, formState.message)
	const { handleSubmit } = useFormSubmission(formValidation.checkInputs, sendEmail)

	return {
		// Form field states
		name: formState.name,
		email: formState.email,
		phone: formState.phone,
		enterprise: formState.enterprise,
		address: formState.address,
		subject: formState.subject,
		message: formState.message,

		// Validation states
		nameValid: formState.nameValid,
		phoneError: null, // No longer tracking phone errors separately
		isLoading: formState.isLoading,

		// Form handlers
		setName: formState.setName,
		setEmail: formState.setEmail,
		setPhone: formState.setPhone,
		setEnterprise: formState.setEnterprise,
		setAddress: formState.setAddress,
		setSubject: formState.setSubject,
		setMessage: formState.setMessage,
		setNameValid: formState.setNameValid,
		setIsLoading: formState.setIsLoading,

		// Validation methods
		validatePhone: formValidation.validatePhone,
		validateForm: formValidation.validateForm,
		resetForm,
		handleSubmit,

		// Validator configurations
		getNameValidators: validatorCreation.getNameValidators,
		getEmailValidators: validatorCreation.getEmailValidators,
		getAddressValidators: validatorCreation.getAddressValidators,
	}
}

// Email sending helper function
const sendContactEmail = (params: {
	name: string | null, email: string | null, phone: string | undefined,
	enterprise: string | null, address: string | null, subject: string | null,
	message: string | null, text: Record<string, string | string[]> | undefined,
	setIsLoading: (loading: boolean) => void,
	resetForm: () => void
}) => {
	const { name, email, phone, enterprise, address, subject, message, text, setIsLoading, resetForm } = params
	const toastLoad = toast.dark(text?.sending)
	setIsLoading(true)
	emailjs
		.send(
			process.env.NEXT_PUBLIC_EMAILJS_SERVICE_ID ?? '',
			process.env.NEXT_PUBLIC_EMAILJS_TEMPLATE_ID ?? '',
			{
					text: message,
					name,
					email,
					phone,
					enterprise,
					address,
					subject,
				},
				process.env.NEXT_PUBLIC_EMAILJS_PUBLIC_KEY ?? '',
			)
			.then(() => {
				toast.dismiss(toastLoad)
				toast.success(text?.messageSent)
				resetForm()
			})
			.catch(() => {
				toast.dismiss(toastLoad)
				toast.error(text?.messageNotSent)
			})
			.finally(() => {
				setIsLoading(false)
			})
}
