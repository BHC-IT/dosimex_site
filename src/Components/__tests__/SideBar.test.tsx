import { render, screen } from '@testing-library/react'
import { useRouter } from 'next/router'
import { describe, it, expect, vi, beforeEach } from 'vitest'

import { useText } from '../../Hooks/useText'
import SideBar from '../SideBar'

// Mock Next.js router
vi.mock('next/router', () => ({
	useRouter: vi.fn(),
}))

// Mock useText hook
vi.mock('../../Hooks/useText', () => ({
	useText: vi.fn(),
}))

// Mock Next.js Image
vi.mock('next/image', () => ({
	default: ({ src, alt, ...props }: any) => (
		<img
			src={src}
			alt={alt}
			{...props}
		/>
	),
}))

// Mock Next.js Link
vi.mock('next/link', () => ({
	default: ({ children, href, ...props }: any) => (
		<a
			href={href}
			{...props}
		>
			{children}
		</a>
	),
}))

// Mock react-burger-menu
vi.mock('react-burger-menu', () => ({
	slide: ({ children, isOpen, ...props }: any) => (
		<div
			data-testid='burger-menu'
			data-is-open={isOpen}
			{...props}
		>
			{children}
		</div>
	),
}))

// Mock components
vi.mock('../Button', () => ({
	default: ({ name }: any) => <button data-testid='sidebar-button'>{name}</button>,
}))

vi.mock('../LanguageSwitch', () => ({
	default: ({ language }: any) => <div data-testid='sidebar-language-switch'>{language}</div>,
}))

vi.mock('../ItemNavbar', () => ({
	default: ({ name }: any) => <div data-testid='nav-item'>{name}</div>,
}))

const mockRouter = {
	push: vi.fn(),
	pathname: '/home',
	locale: 'en',
}

const mockText = {
	items: ['Software', 'Training', 'Manuals', 'About', 'Contact'],
	button: 'Get Started',
}

describe('SideBar Component', () => {
	beforeEach(() => {
		vi.mocked(useRouter).mockReturnValue(mockRouter as any)
		vi.mocked(useText).mockReturnValue(mockText)
		vi.clearAllMocks()
	})

	it('renders burger menu component', () => {
		render(<SideBar />)

		const burgerMenu = screen.getByTestId('burger-menu')
		expect(burgerMenu).toBeInTheDocument()
		// The Menu component receives the right prop, but our mock doesn't need to check it
	})

	it('renders logo with correct properties', () => {
		render(<SideBar />)

		const logo = screen.getByAltText('Dosimex Logo')
		expect(logo).toBeInTheDocument()
		expect(logo).toHaveAttribute('src', '/Images/logo_dosimex_new.webp')
	})

	it('renders all navigation items', () => {
		render(<SideBar />)

		const navItems = screen.getAllByTestId('nav-item')
		expect(navItems).toHaveLength(5)

		expect(screen.getByText('Software')).toBeInTheDocument()
		expect(screen.getByText('Training')).toBeInTheDocument()
		expect(screen.getByText('Manuals')).toBeInTheDocument()
		expect(screen.getByText('About')).toBeInTheDocument()
		expect(screen.getByText('Contact')).toBeInTheDocument()
	})

	it('renders button component', () => {
		render(<SideBar />)

		const button = screen.getByTestId('sidebar-button')
		expect(button).toBeInTheDocument()
		expect(button).toHaveTextContent('Get Started')
	})

	it('renders language switch component', () => {
		render(<SideBar />)

		const languageSwitch = screen.getByTestId('sidebar-language-switch')
		expect(languageSwitch).toBeInTheDocument()
		expect(languageSwitch).toHaveTextContent('en')
	})

	it('passes current pathname and locale to language switch', () => {
		const customRouter = { ...mockRouter, pathname: '/about', locale: 'fr' }
		vi.mocked(useRouter).mockReturnValue(customRouter as any)

		render(<SideBar />)

		const languageSwitch = screen.getByTestId('sidebar-language-switch')
		expect(languageSwitch).toHaveTextContent('fr')
	})

	it('renders logo inside a link to home page', () => {
		render(<SideBar />)

		const logo = screen.getByAltText('Dosimex Logo')
		const link = logo.closest('a')
		expect(link).toHaveAttribute('href', '/')
	})

	it('renders button with correct props from useText', () => {
		const customText = { ...mockText, button: 'Custom Button' }
		vi.mocked(useText).mockReturnValue(customText)

		render(<SideBar />)

		const button = screen.getByTestId('sidebar-button')
		expect(button).toHaveTextContent('Custom Button')
	})

	it('handles empty text items gracefully', () => {
		// Clear the previous mock and set up a new one for this test
		vi.clearAllMocks()
		const emptyText = { items: [], button: 'Test' }
		vi.mocked(useText).mockReturnValue(emptyText)

		const { container } = render(<SideBar />)
		expect(container).toBeInTheDocument()

		// Should not render any nav items when items array is empty
		const navItems = screen.queryAllByTestId('nav-item')
		expect(navItems).toHaveLength(0)
	})

	it('uses correct logo dimensions with ratio', () => {
		render(<SideBar />)

		const logo = screen.getByAltText('Dosimex Logo')

		// Expected dimensions: 212 * 0.5 = 106, 44 * 0.5 = 22
		expect(logo).toHaveAttribute('width', '106')
		expect(logo).toHaveAttribute('height', '22')
	})

	it('accepts text prop and renders consistently', () => {
		const customText = { items: ['Custom', 'Items'], button: 'Custom Button' }
		render(<SideBar text={customText} />)

		const burgerMenu = screen.getByTestId('burger-menu')
		expect(burgerMenu).toBeInTheDocument()

		// The component should still use useText hook regardless of text prop
		const button = screen.getByTestId('sidebar-button')
		expect(button).toHaveTextContent('Get Started') // From mocked useText
	})

	it('should match snapshot with default props', () => {
		const { container } = render(<SideBar />)
		expect(container.firstChild).toMatchSnapshot()
	})

	it('should match snapshot with text prop', () => {
		const testMockText = {
			items: ['Item 1', 'Item 2'],
			button: 'Test Button',
		}
		const { container } = render(<SideBar text={testMockText} />)
		expect(container.firstChild).toMatchSnapshot()
	})
})
