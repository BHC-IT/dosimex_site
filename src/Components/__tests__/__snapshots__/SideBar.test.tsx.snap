// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SideBar Component > should match snapshot with additional props 1`] = `
<div
  customprop="test"
  data-testid="burger-menu"
>
  <ul>
    <li
      style="cursor: pointer;"
    >
      <a
        href="/"
      >
        <img
          alt="logo dosimex"
          height="22"
          loading="lazy"
          quality="40"
          src="/Images/logo_dosimex_new.webp"
          width="106"
        />
      </a>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Software
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Training
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Manuals
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        About
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Contact
      </div>
    </li>
    <div
      data-testid="sidebar-language-switch"
    >
      en
    </div>
  </ul>
  <button
    data-testid="sidebar-button"
  >
    Get Started
  </button>
</div>
`;

exports[`SideBar Component > should match snapshot with default props 1`] = `
<div
  data-testid="burger-menu"
>
  <ul>
    <li
      style="cursor: pointer;"
    >
      <a
        href="/"
      >
        <img
          alt="logo dosimex"
          height="22"
          loading="lazy"
          quality="40"
          src="/Images/logo_dosimex_new.webp"
          width="106"
        />
      </a>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Software
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Training
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Manuals
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        About
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Contact
      </div>
    </li>
    <div
      data-testid="sidebar-language-switch"
    >
      en
    </div>
  </ul>
  <button
    data-testid="sidebar-button"
  >
    Get Started
  </button>
</div>
`;

exports[`SideBar Component > should match snapshot with text prop 1`] = `
<div
  data-testid="burger-menu"
  text="[object Object]"
>
  <ul>
    <li
      style="cursor: pointer;"
    >
      <a
        href="/"
      >
        <img
          alt="logo dosimex"
          height="22"
          loading="lazy"
          quality="40"
          src="/Images/logo_dosimex_new.webp"
          width="106"
        />
      </a>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Software
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Training
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Manuals
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        About
      </div>
    </li>
    <li
      class="menu-item"
    >
      <div
        data-testid="nav-item"
      >
        Contact
      </div>
    </li>
    <div
      data-testid="sidebar-language-switch"
    >
      en
    </div>
  </ul>
  <button
    data-testid="sidebar-button"
  >
    Get Started
  </button>
</div>
`;
